app/models/physical_storage.rb

```diff
@@ -28,6 +28,7 @@ class PhysicalStorage < ApplicationRecord
   has_many :canister_computer_systems, :through => :canisters, :source => :computer_system
   has_many :guest_devices, :through => :hardware
   has_many :wwpn_candidates, :dependent => :destroy
+  has_many :event_streams, :dependent => :destroy
```

Comment:
Prefer to nullify rather than cascade destroy the events, if the physical storage is removed the events could still be helpful.  This is what PhysicalSwitch and PhysicalChassis both do already.

```suggestion
  has_many :event_streams, :dependent => :nullify
```

Reply:
Hi Adam,
I consulted my team leaders and they prefer to destroy the events, since they are not relevant when the physical storage which propagated them is not attached. If the physical storage is re-attached, they are propagated again, so they are not actually lost. 
However, this is not a critical issue.. Do you prefer to nullify rather than destroy anyway?

Reply:
Even if the association to the physical storage record is nulled out the storage name and ems_ref are still there and can still be helpful to the administrator.

We have logic to prevent event duplication so if you add the physical storage back events should not be duplicated.

No other associated inventory cascade destroy events, for consistency please nullify the association.

Reply:
understood,
thanks